__tests__/app.test.js

```diff
@@ -395,4 +395,105 @@ describe('/api/article/:article_id/comments', function(){
             
     });
 
-});
\ No newline at end of file
+});
+
+
+
+
+
+describe('7. POST /api/articles/:article_id/comments', function(){
+
+    describe('201s', function(){
+
+        test('POST 201: responds with a posted json comment object?', function(){
+
+            return superTest(app)
+            .post('/api/articles/9/comments')
+            .send({ username: 'icellusedkars', body: 'This is my comment' })
+            .expect(201)
+            .then((response) => {
+
+                expect(response.body.comment).toBeInstanceOf(Object);
+
+            });
+
+        });
+
+
+        test('POST 201: responds with a posted json comment object?', function(){
+
+            return superTest(app)
+            .post('/api/articles/9/comments')
+            .send({username: 'icellusedkars', body: 'This is my comment'})
+            .expect(201)
+            .then((response) => {
+
+                expect(response.body.comment).toBeInstanceOf(Object);
+
+            });
+
+        });
+
+
+        test('POST 201: has a key of comment with a value of the comment object entered.', function(){
+
+            return superTest(app)
+            .post('/api/articles/9/comments')
+            .send({username: 'icellusedkars', body: 'This is my comment'})
+            .expect(201)
+            .then((response) => {
+
+                expect(response.body.comment).toMatchObject({ 
+
+                    comment_id: expect.any(Number),
+                    article_id: expect.any(Number),
+                    author: expect.any(String), 
+                    body: expect.any(String),
+                    created_at: expect.any(String),
+                    votes: expect.any(Number)
+
+            
+                });                
+
+            });
+
+        });
+
+
+    });
+
+
+    describe('error handling', function(){
+
+        test('POST 404: returns 404 and a message, "not found" when article_id passed in is valid but non existent', function(){
+
+            return superTest(app)
+            .post('/api/articles/7098/comments')
+            .send({username: 'icellusedkars', body: 'This is my comment'})
+            .expect(404)
+            .then((response) => {
+
+                expect(response.body).toEqual({ message: 'not found' });
+
+            });
+
+        });
+
+        test('GET 400: returns 400 and a message, "invalid id" when article_id passed in is not a number', function(){
+
+            return superTest(app)
+            .post('/api/articles/notANumber/comments')
+            .send({username: 'icellusedkars', body: 'This is my comment'})
+            .expect(400)
+            .then((response) => {
+
+                expect(response.body).toEqual({ message: 'invalid id' });
+
+            });
+
+        });
+
```

Comment:
A couple of tests to add: 
201 or 400 - unnecessary properties being added to the comment
400 - missing required keys and values in the sent comment
404 - username doesn't exist on the database